xrpl/models/transactions/amm_bid.py

```diff
@@ -0,0 +1,76 @@
+"""Model for AMMBid transaction type."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional
+
+from typing_extensions import Final
+
+from xrpl.models.amounts import Amount
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+_MAX_AUTH_ACCOUNTS: Final[int] = 4
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AMMBid(Transaction):
+    """
+    AMMBid is used to place a bid for the auction slot of obtaining trading advantages
+    of an AMM instance.
+
+    An AMM instance auctions off the trading advantages to users (arbitrageurs) at a
+    discounted TradingFee for a 24 hour slot.
+    """
+
+    amm_hash: str = REQUIRED  # type: ignore
```

Comment:
Does `AMM` not also need to be added to the special capitalized strings list?

Reply:
Yup, just added it.